docs/specifications/broker-authentication.md

```diff
@@ -0,0 +1,107 @@
+
+The plan for this sprint is to do the following:
+
+* make auth routes configurable by enhancing the configuration
+```
+auth:
+   - type: basic | INSERT OTHER TYPE HERE
+   - enable: true | false
+```
+  * update the following files to reflect the new entry:
+      * [deploy-ansible-service-broker.template.yaml](https://github.com/openshift/ansible-service-broker/blob/master/templates/deploy-ansible-service-broker.template.yaml)
+      * [prep_local_devel_env.sh](https://github.com/openshift/ansible-service-broker/blob/master/scripts/prep_local_devel_env.sh) script
+      * [example-config.yaml](https://github.com/openshift/ansible-service-broker/blob/master/etc/example-config.yaml)
+      * [README.md](https://github.com/openshift/ansible-service-broker/blob/master/README.md)
+      * docs directory:
+          * [deployment.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/deployment.md)
+          * [config.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/config.md)
+          * [local_deployment.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/local_deployment.md)l
+      * [`make deploy`](https://github.com/openshift/ansible-service-broker/blob/master/Makefile#L57-L58) does not need updating because it uses the above template.
+  * [catasb](https://github.com/fusor/catasb/tree/dev)
+      * pulls down the `deploy-ansible-service-broker.template.yaml`
+      * may have to update all.yml in case we need to override the auth setting
+
+* create a secret that both the broker and service catalog can use
+
+* secret rotation is desired but needs more research, see [#open-issues](#open-issues)
+
+* use [`http.BasicAuth`](https://golang.org/pkg/net/http/#Request.BasicAuth) to handle the basic auth case
+  * BasicAuth returns (username, password string, ok bool)
+  * Compare against the values from the secret
+      * if they match, proceed with call
+      * if no match, return [401 Unauthorized](https://golang.org/pkg/net/http/#pkg-constants)
+
+## Future improvements to authentication
+* the [OpenServiceBroker API proposes](https://github.com/openservicebrokerapi/servicebroker/pull/223) making auth optional, but also allow things like [OAuth](https://oauth.net/2/), or other mechanism.
+* the Service Catalog has an issue open to implement [JSON Web Token (JWT)](https://jwt.io/): [issue #990](https://github.com/kubernetes-incubator/service-catalog/issues/990)
+
+## Open Issues
+
+* reloading secrets is different between kubernetes and openshift. [see updating secrets](#updating-secrets)
+    * might need to reconcile the secret on an interval if not automatic
+* shared secrets are mysterious. [OpenShift docs](https://github.com/openshift/openshift-docs/blob/master/dev_guide/secrets.adoc#creating-secrets) state:
+> Secret data can be shared within a namespace
+Except that the broker and the service-catalog run in different namespaces. How
+do we *share* a secret in this case?
```

Comment:
I have same question on how to share the secret between service-catalog/broker 
Probably good to ask @pmorie for some advice here.